nautobot/docs/user-guides/custom-fields.md

```diff
@@ -0,0 +1,323 @@
+![Custom Field Select](./images/custom-fields/custom_field_select.png)
+
+## Editing Custom Fields
+
+Since automatic provisioning is only performed when a custom field's content types change, some changes made to existing custom fields are not reflected on the associated objects automatically. Some examples of cases where this might cause unexpected behavior are changes to the `required`, `default` and validation fields.
+
+If a custom field is created with `required=False` and then later changed to `required=True`, all of the associated objects that do not yet have a value for this field will fail validation the next time they're saved unless updated with a valid value. Similarly, changes to the validation fields for a custom field will not automatically result in changes to affected objects, but may require changes the next time those objects are edited in order to bring them into compliance with the updated validation rules.
+
+If a custom field's `default` value is changed, newly created objects will use the new default, but existing objects that were set to the previous default value will *not* automatically be updated.
+
+## Deleting Custom Fields
+
+Custom fields are removed from associated objects when a content type is removed from the custom field, including when the custom field is deleted. This update runs as a background task via [Celery](../installation/services.md#worker-service), so it may take a few seconds or more before the custom field data is removed from all objects, depending on the size of your database.
+
+## Retrieving Custom Field Data
+
+Custom fields augment an existing model so retrieving custom field values is different from native fields. All custom field data is stored as a dictionary in the model field named `_custom_field_data` but there is a property named `cf` to make accessing this field easier. Example:
+
+### Retrieve Custom Field Data in Nautobot Shell
+
+```py
+# retrieve all custom field data
+>>> device.cf
+{'eol_date': '1970-01-01', 'support_group': 'Network Operations (555-4357)', 'dmz_device': True, 'cmdb_id': 12345}
+
+# retrieve a single field
+>>> device.cf.get("eol_date")
+'1970-01-01'
+```
+
+The `slug` of the custom field is used as the key for the associated object's data dictionary.
+
+### Retrieve Custom Field Data in the Rest API
+
+Custom fields are returned in the API for all supported models in the `custom_fields` key:
+
+??? info "`GET http://localhost:8080/api/dcim/devices/ffd8df99-6d1a-41c3-b19f-b8357eefc481/`"
+
+    ```json
+    {
+      "id": "ffd8df99-6d1a-41c3-b19f-b8357eefc481",
+      ...
+      "custom_fields": {
+        "dmz_device": true,
+        "eol_date": "1970-01-01",
+        "cmdb_id": 12345,
+        "support_group": "Network Operations (555-4357)"
+      }
+    }
+    ```
+
+### GraphQL
+
+#### Retrieve Data for a Custom Field in GraphQL
+
+Individual custom fields can be retrieved in GraphQL queries by using the `cf_<fieldname>` field name format:
+
+```graphql
```

Comment:
Unfortunately it doesn't look like the `graphql` syntax is highlighting. Is there another plugin we need to install/enable?

Reply:
It doesn't look like graphql is supported in the code highlighting library that material for mkdocs uses, [`pygments`](https://pygments.org/languages/).  #2571